sdk/core/azure-core-test/src/main/java/com/azure/core/test/http/PlaybackClient.java

```diff
@@ -68,11 +73,7 @@ public PlaybackClient(RecordedData recordedData, Map<String, String> textReplace
 
             return Mono.error(new IllegalStateException("==> Unexpected request: " + incomingMethod + " " + incomingUrl));
         }
-
-        if (networkCallRecord.exception() != null) {
-            throw logger.logExceptionAsWarning(Exceptions.propagate(networkCallRecord.exception().get()));
-        }
-
+        constructExceptionType(networkCallRecord);
```

Comment:
You should change this. At present the `constructExceptionType` call is odd, as it may throw an exception but it won't if the `networkCallRecord.exception()` is null. You should reinstitute the if block you removed, and then make constructExceptionType return an exception that is then definitively thrown.

Reply:
the step only takes affect when exception is not null. 
Will add a checking before calling the method.
Thanks for catching this.

Reply:
My suggestion and what you implemented are slightly different. Please re-read and clarify with me if necessary.